logic.js

```diff
@@ -0,0 +1,195 @@
+//  a module to serve as our database and useful methods
+
+
+const Tweeter = function () {
+    const _posts = [  // posts and comments data base
+        {
+            id: "p1",
+            text: "First post!",
+            comments: [
+                { id: "c1", text: "First comment on first post!" },
+                { id: "c2", text: "Second comment on first post!!" },
+                { id: "c3", text: "Third comment on first post!!!" }
+            ]
+        },
+        {
+            id: "p2",
+            text: "Aw man, I wanted to be first",
+            comments: [
+                { id: "c4", text: "Don't wory second poster, you'll be first one day." },
+                { id: "c5", text: "Yeah, believe in yourself!" },
+                { id: "c6", text: "Haha second place what a joke." }
+            ]
+        }
+    ]
+
+
+// available methods of Tweeter 
+
+
+let postsCounter = _posts.length+1  // counter for number of posts already stored
```

Comment:
A counter should never be initialized via array length, it's can cause duplication of the same id (for example, if few posts where deleted, the length is back to the previous length)